Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -26,6 +27,8 @@
  */
 abstract class AbstractFilter implements FilterInterface
 {
+    use RequestTrait;
```

Comment:
I concur with @dunglas that traits enable horizontal composition. There isn't really any better alternative in PHP.

The second part of your above comment (which is oddly contradictory to the first part) has only inspired me to extract those functions into a separate trait to allow greater reuse outside of `AbstractFilter` (for custom filters which are not extending it).